src/Tickets/Commerce/Gateways/Stripe/Webhooks.php

```diff
@@ -227,6 +246,150 @@ public function handle_validation(): void {
+
+		return true;
+	}
+
+	/**
+	 * Disables the current webhook.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function disable_webhook() {
+		if ( ! $this->get_gateway()->is_active() ) {
+			// Bail if stripe is not active.
+			return false;
+		}
+
+		if ( ! $this->has_valid_signing_secret() ) {
+			// If we don't have a webhook set up and validated we won't disable any.
+			return false;
+		}
+
+		// Pinpoint the current webhook in use.
+		$known_webhooks = tribe_get_option( self::OPTION_KNOWN_WEBHOOKS, [] );
+
+		$current_signing_key = tribe_get_option( static::$option_webhooks_signing_key );
+
+		$known_webhooks = array_filter(
+			$known_webhooks,
+			function ( $signing_key ) use ( $current_signing_key ) {
+				return $signing_key === $current_signing_key;
+			}
+		);
+
+		// Current being used, not known. We bail.
+		if ( empty( $known_webhooks ) ) {
+			return false;
+		}
+
+		$webhook_disable_endpoint = tribe( WhoDat::class )->get_api_url(
+			'webhook/disable',
+			[
+				'stripe_user_id' => rawurlencode( tribe( Merchant::class )->get_client_id() ),
+				'home_url'       => rawurlencode( tribe( Return_Endpoint::class )->get_route_url() ),
+				'version'        => rawurlencode( \Tribe__Tickets__Main::VERSION ),
+				'known_webhooks' => array_map( 'rawurlencode', array_keys( $known_webhooks ) ),
+			]
+		);
+
+		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get
+		$response = wp_remote_get( $webhook_disable_endpoint );
+
+		$body = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		if ( empty( $body['webhook']['id'] ) ) {
+			return false;
+		}
+
+		// Invalidate webhook related options.
+		tribe_update_option( self::$option_webhooks_signing_key, '' );
+		tribe_update_option( self::$option_is_valid_webhooks, false );
+
+		return true;
+	}
+
+	/**
+	 * Adds a webhook to the known webhooks and updates the current signing secrets
+	 * and marks it as validated.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $webhook The webhook being added.
+	 * @return void
+	 */
+	public function add_webhook( $webhook ) {
+		$known_webhooks = tribe_get_option( self::OPTION_KNOWN_WEBHOOKS, [] );
+
+		$signing_key = $webhook['secret'] ?? false;
+		$signing_key = $signing_key ? $signing_key : ( $known_webhooks[ $webhook['id'] ] ?? false );
+
+		$known_webhooks[ $webhook['id'] ] = $signing_key;
+		// Keeping multiple and unlimited known_webhooks is not a good idea.
+		// If abused it could grow too big into the DB and also cause the whodat URLs to become longer than 2048 chars which is the limit.
+		// In any case, each installation should have only one webhook.
+		// Let's keep a maximum of 3 known webhooks.
+		if ( count( $known_webhooks ) > 3 ) {
+			foreach ( array_keys( $known_webhooks ) as $key ) {
+				if ( $key === $webhook['id'] ) {
+					continue;
+				}
+
+				// Unset a random one but not the one we just added.
+				unset( $known_webhooks[ $key ] );
+				break;
+			}
+		}
```

Comment:
On application level we are ensuring since the check is constant. Meaning everytime that the $known_webhooks would grow larger than 3 (e.g. 4) it removes one. But you are right that an array_slice solution would be more consistent and ensuring this was the case even when somebody added a value directly in the DB.

Reply:
Implemented on the latest commit @codingmusician !